kedro/framework/cli/starters.py

```diff
@@ -321,6 +328,7 @@ def new(config_path, starter_alias, selected_addons, checkout, directory, **kwar
         config = _fetch_config_from_user_prompts(prompts_required, cookiecutter_context)
 
     config = _get_addons_from_cli_input(selected_addons, config)
+    config = _get_name_from_cli_input(name, config)
```

Comment:
It's generally not good practice to modify your input, in this case `config`, in this way. Here it's also getting hard to follow what "config" is exactly. I'd probably change this to something like: 

```suggestion
    config_with_add_ons = _get_addons_from_cli_input(selected_addons, config)
    config_with_add_ons_and_name = _get_name_from_cli_input(name, config_with_add_ons)
```